src/controllers/userController.js

```diff
@@ -43,6 +44,63 @@ const signUp = async (req, res) => {
   }
 };
 
+/**
+ * get user details by id
+ * @param {Object} req - server request
+ * @param {Object} res - server response
+ * @returns {Object} - custom response
+ * @description get's details of logged in user
+ */
+const getUserDetailsById = async (req, res) => {
+  try {
+    const { userId } = req.params;
+    const options = { attributes: { exclude: ['password'] } };
+    const user = await getById(User, userId, options);
+    return response(res, 200, 'success', { user });
+  } catch (error) {
+    return response(res, 500, 'error', { message: 'Internal server error' });
+  }
+};
+
+/**
+ * update user details
+ * @param {Object} req - server request
+ * @param {Object} res - server response
+ * @returns {Object} - custom response
+ * @description updates details of logged in user
+ */
+const updateUserDetails = async (req, res) => {
+  try {
+    const { userId } = req.params;
+    const { id } = req.decoded;
+    if (userId !== id) {
+      return response(res, 403, 'error', { message: 'You are not authorized to edit this profile' });
+    }
+    const {
+      firstName, lastName, phoneNo, birthDate, preferredLanguage,
+      preferredCurrency, gender, role, lineManager, currentLocation
+    } = req.body;
+    const userInfo = {
+      firstName,
+      lastName,
+      phoneNo,
+      birthDate,
+      preferredLanguage,
+      preferredCurrency,
+      gender,
+      role,
+      lineManager,
+      currentLocation
+    };
+    const updatedUser = await update(User, userId, userInfo);
```

Comment:
This update function is not descriptive enough

Reply:
can you please shed more light on this.